src/pvrclient-nextpvr.cpp

```diff
@@ -743,45 +762,58 @@ bool cPVRClientNextPVR::CanSeekStream(void)
 bool cPVRClientNextPVR::OpenRecordedStream(const kodi::addon::PVRRecording& recording, int64_t& streamId)
 {
   kodi::addon::PVRRecording copyRecording = recording;
-  m_nowPlaying = Recording;
   copyRecording.SetDirectory(m_recordings.m_hostFilenames[recording.GetRecordingId()]);
   const std::string line = kodi::tools::StringUtils::Format("%s/live?recording=%s&client=XBMC-%s", m_settings->m_urlBase, recording.GetRecordingId().c_str(), m_request.GetSID());
-  return m_recordingBuffer->Open(line, copyRecording);
+  m_mutexMulti.lock();
+  m_nowPlaying = Recording;
+  m_multistreamRecording.emplace(++m_streamCount, new timeshift::RecordingBuffer(m_settings, m_request));
+  streamId = m_streamCount;
+  bool ret = m_multistreamRecording[streamId]->Open(line, copyRecording, streamId);
+  if (!ret)
+  {
+    CloseRecordedStream(streamId);
+  }
+  m_mutexMulti.unlock();
+  return ret;
 }
 
 void cPVRClientNextPVR::CloseRecordedStream(int64_t streamId)
 {
-  if (IsServerStreamingRecording())
+  if (IsServerStreamingRecording(streamId))
   {
-    m_recordingBuffer->Close();
-    m_recordingBuffer->SetDuration(0);
+    m_mutexMulti.lock();
+    m_multistreamRecording[streamId]->Close();
+    m_multistreamRecording.erase(streamId);
+    m_mutexMulti.unlock();
   }
-  m_nowPlaying = NotPlaying;
+  if (m_multistreamRecording.size() == 0)
```

Comment:
`std::lock_guard<std::recursive_mutex> lock(m_multistreamRecording);` prior to accessing map.

No need to stop the lock, happens automatically at the end block